### PR TITLE
Update versioning in setup.py

### PR DESCRIPTION
**Pull request by sumith1896**
Following NumPy, a much cleaner approach

Development version

```
>>> import sympy as sp
>>> sp.__version__
'0.7.6.1.dev0+9c57429'
```

Release version

```
>>> import sympy as sp
>>> sp.__version__
'0.7.6.1'
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -308,12 +314,84 @@ def run(self):
...
+    try:
+        a.write(cnt % {'version': VERSION,
+                       'full_version' : FULLVERSION,
+                       'git_revision' : GIT_REVISION,
+                       'isrelease': str(ISRELEASED)})
+    finally:
+        a.close()
+
+write_version_py()
+from sympy.version import version as __version__
```
Comment (certik) [inline setup.py]: I think you cannot import sympy from setup.py. I forgot why.

**Comment by certik** [comment]
Travis tests fail.

File: setup.py [hunk 2]
```diff
@@ -308,12 +314,84 @@ def run(self):
...
+
 with open('sympy/__init__.py') as f:
     long_description = f.read().split('"""')[1]
 
 setup(name='sympy',
-      version=__version__,
+	version=__version__,
```
Comment (Sumith1896) [inline setup.py]: @certik Is this needed?
NumPy doesn't seem to have it.

Comment (certik) [inline setup.py]: I think this is needed, so that you can retrieve the version using setuptools.

**Comment by Sumith1896** [comment]
This will require some re-factoring as @isuruf was suggesting. I'll look into that.
I hope the idea is fine.

File: setup.py (hunk 1 above)
Comment (moorepants) [inline setup.py]: You can do this: https://github.com/pydy/pydy/blob/master/setup.py#L9

**Comment by moorepants** [comment]
On the pep 440 change we decided to use `0.7.7.dev` instead of `0.7.7.dev0`. The `0` is not necessary for us.

File: setup.py (hunk 1 above)
Comment (Sumith1896) [inline setup.py]: https://github.com/symengine/symengine.py/pull/6#discussion_r42395232

**Comment by asmeurer** [comment]
This seems to complicate things in my view. What was wrong with the way it was before?

**Comment by asmeurer** [comment]
-1 to this. If I import sympy, I get

```
Traceback (most recent call last):
  File "./bin/isympy", line 357, in <module>
    main()
  File "./bin/isympy", line 353, in main
    from sympy.interactive import init_session
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/__init__.py", line 21, in <module>
    from .version import git_revision as __git_revision__
ImportError: No module named 'sympy.version'
```

You shouldn't have to run setup.py just to be able to import SymPy!

**Comment by asmeurer** [comment]
If we really want to do git versioning, we should use versioneer. But I don't really see why it's needed. The advantage is that if someone is using the development version, then we can see exactly what commit they are using from the version, but the disadvantage is that the version is taken from the git tag, which potentially makes the release process a little more complicated (at least in the sense that you can't fully test a release until after you make a tag).

**Comment by certik** [comment]
I originally thought that `exec(open('sympy/release.py').read())` is ugly. And it is. But now when I see the other option, it seems it's actually a very simple solution. Sorry @Sumith1896 for thinking that this could be easier. Looks like when all things are considered, the original code is simpler.

**Comment by moorepants** [comment]
Yes, the `exec` call is ugly, but after researching this a while back, it seems like the simplest solution.

**Comment by moorepants** [comment]
I'm +1 for having the git commit hash in the development version number. It will help significantly when we deal with bugs in development branches and improve the communication. We won't have to ask explicit for people to retrieve the commit hash.

**Comment by Sumith1896** [comment]
I'm +1 for the git version but even I feel that the `setup.py` has a lot of code that is doing very little of significance also the importing SymPy problem @asmeurer stated, that is a serious. It went off my mind.

**Comment by asmeurer** [comment]
If we really want the git version, the best way to do it is to use versioneer.

**Comment by Sumith1896** [comment]
This is not the way forward. We'll need to use versioneer if we need git revision.

Anyways, thanks @certik @moorepants @asmeurer for the insights!

**Pull request closed without merging**